File: Prog2-9302-AY225-Salvador/MIDTERM-LAB-3/MP11_ColumnFrequency/MP11_ColumnFrequency.js
const fs = require('fs');

const filePath = "../data.csv"; // CSV in parent folder
const columnName = "Exam";      // change this to any column you want

fs.readFile(filePath, 'utf8', (err, data) => {
    if (err) {
        console.log('Error reading CSV:', err.message);
        return;
    }

    const lines = data.trim().split('\n');
    const headers = lines[0].split(',');
    const columnIndex = headers.findIndex(h => h.trim().toLowerCase() === columnName.trim().toLowerCase());

    if (columnIndex === -1) {
        console.log('Column not found.');
        return;
    }

    const frequency = {};
    for (let i = 1; i < lines.length; i++) {
        const values = lines[i].split(',');
        const key = values[columnIndex] ? values[columnIndex].trim() : '';
        frequency[key] = (frequency[key] || 0) + 1;
    }

    console.log(`Frequency for column "${columnName}":`);
    for (const [key, count] of Object.entries(frequency)) {
        console.log(`${key} : ${count}`);
    }
});

File: MIDTERM-LAB-3/MP11_ColumnFrequency/MP11_ColumnFrequency.java
import java.io.*;
import java.util.*;

public class MP11_ColumnFrequency {
    public static void main(String[] args) {
        String filePath = "../data.csv"; // CSV in parent folder
        String columnName = "Exam";      // change this to any column you want

        try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {
            String headerLine = br.readLine();
            if (headerLine == null) {
                System.out.println("CSV is empty.");
                return;
            }

            String[] headers = headerLine.split(",");
            int columnIndex = -1;
            for (int i = 0; i < headers.length; i++) {
                if (headers[i].trim().equalsIgnoreCase(columnName.trim())) {
                    columnIndex = i;
                    break;
                }
            }
            
            if (columnIndex == -1) {
                System.out.println("Column not found.");
                return;
            }

            Map<String, Integer> frequency = new HashMap<>();
            String line;
            while ((line = br.readLine()) != null) {
                String[] values = line.split(",");
                if (values.length > columnIndex) {
                    String key = values[columnIndex].trim();
                    frequency.put(key, frequency.getOrDefault(key, 0) + 1);
                }
            }

            System.out.println("Frequency for column \"" + columnName + "\":");
            for (Map.Entry<String, Integer> entry : frequency.entrySet()) {
                System.out.println(entry.getKey() + " : " + entry.getValue());
            }

        } catch (IOException e) {
            System.out.println("Error reading CSV: " + e.getMessage());
        }
    }
}